### DISCUSSION TITLE
Multivectors - K-Vector Element Access

### DISCUSSION THREAD

**Discussion by CaseySanchez** [laffernandes/gatl#28] [Ideas]
Hello Dr. Fernandes,

I have found this paper which the author claims generalizes decompositions of spinors, titled "SWING-TWIST DECOMPOSITION IN CLIFFORD ALGEBRA":
https://arxiv.org/pdf/1506.05481.pdf

In it he uses the Hodge star operator upon the bivector component of the spinor, which appears as follows:
`\star \langle R \rangle_{2} = - \mathbf{I} \langle R \rangle_{2}`

I could see there being value in allowing users to access individual k-vector components of multivectors with the `operator[]` overload. For that matter, it would probably be useful to have a `hodge` function as well.

Regards,
Casey

**Comment by CaseySanchez** [comment]
It appears that the inverse of the Hodge operator is as follows:

```
\begin{equation*}
\star ^{-1} = 
\begin{cases}
s \star & n \text{ is odd} \\
(-1)^{k} s \star & n \text{ is even}
\end{cases}
\end{equation*}
```

**Reply by laffernandes** [reply]
What you are looking for is the `take_grade(arg, k)` function implemented in GATL. You can set the grade `k` as a runtime or a compile-time defined integer value. For the latter case, use `take_grade(r, c<2>)`.
